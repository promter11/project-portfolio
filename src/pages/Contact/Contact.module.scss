@import "../../styles/_variables";

.contact {
  height: 100%;
  padding: 8rem 14rem;
  background: rgba(0, 0, 0, 0.75) url("../../assets/images/Contact.jpg")
    no-repeat center center / cover;
  background-blend-mode: darken;
}

.title {
  margin-bottom: 8rem;
  text-align: right;
  letter-spacing: 1rem;
  opacity: 0.25;
}

.content {
  color: $white-color;
}

.contentTitle {
  margin-bottom: 2rem;
  font-size: 2.4rem;
  font-weight: 900;
  line-height: 2.8rem;
  text-transform: uppercase;
}

.wrapper {
  display: flex;
}

.block {
  padding: 4rem;
  background-color: $white-color;

  &.blue {
    background-color: $contact-block-color;
  }

  &:first-child {
    width: 40%;
    border-radius: 0.4rem 0 0 0.4rem;
  }

  &:last-child {
    width: 60%;
    border-radius: 0 0.4rem 0.4rem 0;
  }
}

.blockTitle {
  margin-bottom: 8rem;
  font-size: 1.8rem;
  font-weight: 900;
  line-height: 2.4rem;

  &.black {
    color: $black-color;
  }
}

.data {
  display: flex;
  align-items: center;

  &:not(:last-child) {
    margin-bottom: 3rem;
  }
}

.dataImage {
  display: block;
  width: 3rem;
  height: 3rem;
  margin-right: 1.5rem;
}

.dataText {
  font-size: 1.4rem;
  font-weight: 100;
  line-height: 1.8rem;
  color: $white-color;
}

.form {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  align-items: flex-start;
}

.formLabel {
  width: calc(50% - 1.5rem);
  margin-bottom: 3rem;

  &.formLabelFull {
    width: 100%;
  }
}

.formText {
  display: block;
  margin-bottom: 1rem;
  font-size: 1.4rem;
  font-weight: bold;
  line-height: 1;
  text-transform: uppercase;
  color: $black-color;
}

.formInput,
.formTextarea {
  width: 100%;
  transition: border 0.25s ease;

  &:hover {
    border-bottom: 0.1rem solid lighten($contact-block-color, 25%);
  }

  &:focus {
    border: 0.1rem solid $contact-block-color;
  }
}

.formInput,
.formTextarea,
.formSubmit {
  font-family: "Public Sans", sans-serif;
  font-size: 1.4rem;
  font-weight: normal;
  line-height: 150%;
  padding: 1rem 1.5rem;
  border: 0.1rem solid $form-border-color;
  outline: none;
  background-color: transparent;
  color: $black-color;
}

.formTextarea {
  height: 12rem;
  resize: none;
}

.formSubmit {
  line-height: 1;
  font-weight: bold;
  text-transform: uppercase;
  padding: 1.5rem 3rem;
  border: 0.1rem solid $black-color;
  cursor: pointer;
  transition: all 0.5s ease;
  background-color: transparent;
  color: $black-color;

  &:hover {
    box-shadow: inset 6rem 0 0 0 $contact-block-color,
      inset -6rem 0 0 0 $contact-block-color;
    background-color: transparent;
    color: $white-color;
  }

  &:active {
    transition: none;
    transform: translateY(0.2rem);
  }
}
