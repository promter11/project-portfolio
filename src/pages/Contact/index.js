import React from "react";

import Pin from "../../assets/icons/contact/pin.svg";
import Phone from "../../assets/icons/contact/phone-call.svg";
import Email from "../../assets/icons/contact/email.svg";
import styles from "./Contact.module.scss";

const Contact = () => {
  return (
    <section className={styles.contact}>
      <h1 className={styles.title}>Contact</h1>
      <div className={styles.content}>
        <h2 className={styles.contentTitle}>Let's keep in touch</h2>
        <div className={styles.wrapper}>
          <div className={`${styles.block} ${styles.blue}`}>
            <h3 className={styles.blockTitle}>Contacts</h3>
            <div className={styles.data}>
              <img className={styles.dataImage} src={Pin} alt="Pin" />
              <span className={styles.dataText}>Volgograd, Russia</span>
            </div>
            <div className={styles.data}>
              <img className={styles.dataImage} src={Phone} alt="Phone" />
              <span className={styles.dataText}>+7 (937) 098-05-94</span>
            </div>
            <div className={styles.data}>
              <img className={styles.dataImage} src={Email} alt="Email" />
              <span className={styles.dataText}>promter11@gmail.com</span>
            </div>
          </div>
          <div className={styles.block}>
            <h2 className={`${styles.blockTitle} ${styles.black}`}>
              Send me a message
            </h2>
            <form className={styles.form}>
              <label className={styles.formLabel}>
                <span className={styles.formText}>Name *</span>
                <input
                  className={styles.formInput}
                  type="text"
                  maxLength={40}
                />
              </label>
              <label className={styles.formLabel}>
                <span className={styles.formText}>E-Mail *</span>
                <input
                  className={styles.formInput}
                  type="text"
                  maxLength={40}
                />
              </label>
              <label className={styles.formLabel}>
                <span className={styles.formText}>Message *</span>
                <textarea className={styles.formTextarea} maxLength={200} />
              </label>
              <button className={styles.formSubmit}>Submit</button>
            </form>
          </div>
        </div>
      </div>
    </section>
  );
};

export default Contact;
