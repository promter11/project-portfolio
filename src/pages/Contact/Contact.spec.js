import React from "react";
import { shallow } from "enzyme";

import Contact from "./index";

const props = {
  language: "RU",
};

describe("Contact page", () => {
  it("renders correctly with props", () => {
    const wrapper = shallow(<Contact {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
