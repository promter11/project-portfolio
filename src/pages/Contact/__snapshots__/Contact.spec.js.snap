// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact page renders correctly with props 1`] = `
<section
  className="contact"
>
  <Container>
    <Heading
      className="title"
      level={1}
    >
      Контакты
    </Heading>
    <div
      className="content"
    >
      <Heading
        className="contentTitle"
        level={2}
      >
        Давайте оставаться на связи
      </Heading>
      <div
        className="wrapper"
      >
        <div
          className="block blue"
        >
          <Heading
            className="blockTitle"
            level={3}
          >
            Контактные данные
          </Heading>
          <Info
            items={
              Array [
                Object {
                  "image": "pin.svg",
                  "value": "Волгоград, Россия",
                },
                Object {
                  "image": "phone-call.svg",
                  "path": "tel:79370980594",
                  "value": "+7 (937) 098-05-94",
                },
                Object {
                  "image": "email.svg",
                  "path": "mailto:promter11@gmail.com",
                  "value": "promter11@gmail.com",
                },
              ]
            }
          />
        </div>
        <div
          className="block"
        >
          <Heading
            className="blockTitle black"
            level={3}
          >
            Отправьте мне сообщение
          </Heading>
          <ContactForm
            language="RU"
          />
        </div>
      </div>
    </div>
  </Container>
</section>
`;
